animations/shared-element-transitions/index.md

```diff
@@ -0,0 +1,684 @@
+route to another. This guide describes the first example.<br><br>
+
+[flying_transition](https://github.com/flutter/website/tree/master/_includes/code/animation/flying_transition)
+: Movement along a curved path, as described in the Material motion
+  spec. Encapsulates the Hero code in a custom PhotoHero widget.
+
+[basic_flying_transition](https://github.com/flutter/website/tree/master/_includes/code/animation/basic_flying_transition)
+: Uses the Hero object directly.
+  This more basic example, provided for your reference, isn't
+  described in this guide.
+
+</aside>
+
+Flying an element from one route to another, one of the most
+common types of shared element transitions, is easy to implement
+using Flutter's Hero widget. When using MaterialPageRoute
+to specify the new route, the element flies along a curved path,
+as described by the [Material Design motion
+spec.](https://material.io/guidelines/motion/movement.html)
+The widget tree for the destination route is created in the `builder`
+property.
+
+[Create a new Flutter example](/getting-started/) and
+update it using the files from the
+[GitHub directory.](https://github.com/flutter/website/tree/master/_includes/code/animation/flying_transition)
+
+To run the example:
+
+* Tap on the home route’s photo to fly the image to a new route
+  showing the same photo at a different location and scale.
+* Return to the previous route by tapping the image, or by using the
+  device’s back-to-the-previous-screen gesture.
+* You can slow the transition further using the `timeDilation`
+  property.
+
+### PhotoHero class
+
+The custom PhotoHero class maintains the Hero, and its size, image,
+and behavior when tapped. The PhotoHero builds the following
+widget tree:
+
+<img src="images/photohero-class.png" alt="widget tree for the PhotoHero class">
+
+Here's the code:
+
+<!-- skip -->
+{% prettify dart %}
+class PhotoHero extends StatelessWidget {
+  const PhotoHero({ Key key, this.photo, this.onTap, this.width }) : super(key: key);
+
+  final String photo;
+  final VoidCallback onTap;
+  final double width;
+
+  Widget build(BuildContext context) {
+    return new SizedBox(
+      width: width,
+      child: new Hero(
+        tag: photo,
+        child: new Material(
+          color: Colors.transparent,
+          child: new InkWell(
+            onTap: onTap,
+            child: new Image.asset(
+              photo,
+              fit: BoxFit.contain,
+            ),
+          ),
+        ),
+      ),
+    );
+  }
+}
+{% endprettify %}
+
+Key information:
+
+* An InkWell wraps the Hero, making it trivial to add a tap
+  gesture to the both the source and destination Heroes.
+* Defining the Material widget with a transparent color
+  enables the image to "pop out" of the background as it
+  flies to its destination.
+* Placing everything in a SizedBox and setting it's width
+  to the passed-in value, determines the size
+  of the image at the end of the animation.
```

Comment:
Really it's the Hero's size at start and end of the animation. A PhotoHero is created for the source and Destination routes.